### PR TITLE
[NDM] Add mock Cisco SDWAN API server

### PR DESCRIPTION
**Pull request by TCheruy**
### What does this PR do?
Adds a mock API server
Useful for integration testing of the Cisco SD-WAN API

### Motivation
NDMII-2575

### CODE REVIEW

**Comment by TCheruy** [comment]
### Current dependencies :
- [[NDM] Add mock Cisco SDWAN API server](https://github.com/DataDog/datadog-agent/pull/24094)
  - [[NDM] Add Cisco SD-WAN payload processor](https://github.com/DataDog/datadog-agent/pull/24095)
     -  [[NDM] Add Cisco SD-WAN metrics and metadata sender](https://github.com/DataDog/datadog-agent/pull/24101)
         - [[NDM] Add Cisco SD-WAN corecheck](https://github.com/DataDog/datadog-agent/pull/24102)

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (client *Client) get(endpoint string, params map[string]string) ([]byte, er
 		}
 	}
 
-	return nil, fmt.Errorf("http responsed with %d code", statusCode)
+	return nil, fmt.Errorf("%s http responsed with %d code", endpoint, statusCode)
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go:85]: Unrelated language nit 😛 
```suggestion
	return nil, fmt.Errorf("%s http responded with %d code", endpoint, statusCode)
```

**Review by FlorianVeaux — APPROVED** [review]

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go (hunk 1 above)
Comment (TCheruy) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go]: thanks!

**Review by FlorianVeaux — APPROVED** [review]

**Pull request merged**